### PR TITLE
Add hide-cursor command line argument & always hide cursor option

### LINKED ISSUE #4445: [Avalonia Build] HideCursor arguments
### Overview

Request to extend the following arguments from #4193 to Avalonia Build to override(bypass) the Config.json setting when a game is launched in command line.

- HideCursor
  - default: OnIdle
  - possible values: Never, OnIdle, Always

### Smaller details

User can choose the mouse cursor comportment in the command line at game launch.

### Nature of request

Additionnal arguments for Avalonia Build.

### Why would this feature be useful?

Same reason and use case as here: https://github.com/Ryujinx/Ryujinx/pull/4193#issuecomment-1373376789

### PR DESCRIPTION
**Pull request by TSRBerry**
This PR adds `--hide-cursor` as a new command line argument which overrides the current settings.
Possible values are: "Always", "OnIdle", "Never"

The default value for the hide-cursor option will still be "Never".

---

Closes #4445

### CODE REVIEW

**Comment by jojodunet** [comment]
Tests from this build:
```
ava-ryujinx-Release-1.1.0+90a9d95-win_x64.zip
```

First, I tested the options from the AVA GUI only: 
```
Hide Cursor: Never/OnIdle/Always 
It worked as expected.
```

Afterwards, I tested it using the commande line : 

```
Set in the AVA GUI => Hide Cursor: Never
> Ruyjinx.Ava.exe -f -hide-cursor Always Game.nsp
It doesn't work, Hide-Cursor uses the value "Never" as set in the AVA GUI
```
```
Set in the AVA GUI => Hide Cursor: Always
> Ruyjinx.Ava.exe -f -hide-cursor Never Game.nsp
It doesn't work, Hide-Cursor uses the value "Always" as set in the AVA GUI
```
I haven't tried the command line with OnIdle, but I don't think it works like the other values above.

I noticed some other mistakes:
- The translation of some words is missing according to the language setting in the AVA GUI.
- The correct argument is -hide-cursor (not --hide-cursor).

I hope this helps !

Trad Pic
Error

**Comment by TSRBerry** [comment]
Thank you for your detailed test report, I'll fix these issues and let you know when a new build is available.

**Comment by TSRBerry** [comment]
> I noticed some other mistakes:
>
>    The translation of some words is missing according to the language setting in the AVA GUI.
>    The correct argument is -hide-cursor (not --hide-cursor).

I can't do the translation for other languages, but you can add them on crowdin as soon as this is merged.

`--hide-cursor` is the correct option, but I made another mistake in the Ava version which caused the command line option to not work as expected. But that should be fixed now!

**Comment by jojodunet** [comment]
Tests from this build:
```
ava-ryujinx-Release-1.1.0+ef6985a-win_x64.zip
```

First, I tested the options from the AVA GUI only: 
```
Hide Cursor: Never/OnIdle/Always 
It worked as expected.
```

Afterwards, I tested it using the command line : 

```
Set in the AVA GUI => Hide Cursor: Never
Run > Ruyjinx.Ava.exe -f --hide-cursor Always Game.nsp
Run > Ruyjinx.Ava.exe -f --hide-cursor OnIdle Game.nsp
It worked as expected.
```
```
Set in the AVA GUI => Hide Cursor: Always
Run > Ruyjinx.Ava.exe -f --hide-cursor Never Game.nsp
Run > Ruyjinx.Ava.exe -f --hide-cursor OnIdle Game.nsp
It worked as expected.
```
```
Set in the AVA GUI => Hide Cursor: OnIdle
Run > Ruyjinx.Ava.exe -f --hide-cursor Never Game.nsp
Run > Ruyjinx.Ava.exe -f --hide-cursor Always Game.nsp
It worked as expected.
```

Great work !

For the missing translation, I'll go to Crowdin to complete the translation.

File: Ryujinx/Ui/Windows/SettingsWindow.glade [hunk 1]
```diff
@@ -161,19 +160,86 @@
...
+                                        <property name="position">2</property>
+                                      </packing>
+                                    </child>
+                                    <child>
+                                      <object class="GtkRadioButton" id="_hideCursorNever">
+                                        <property name="label" translatable="yes">Never</property>
+                                        <property name="visible">True</property>
+                                        <property name="can-focus">True</property>
+                                        <property name="receives-default">False</property>
+                                        <property name="tooltip-text" translatable="yes">Use a software page table for address translation. Highest accuracy but slowest performance.</property>
```
Comment (riperiperi) [inline Ryujinx/Ui/Windows/SettingsWindow.glade:186]: Tooltip is incorrect.

File: Ryujinx/Ui/Windows/SettingsWindow.glade [hunk 2]
```diff
@@ -161,19 +160,86 @@
...
+                                        <property name="position">3</property>
+                                      </packing>
+                                    </child>
+                                    <child>
+                                      <object class="GtkRadioButton" id="_hideCursorOnIdle">
+                                        <property name="label" translatable="yes">On Idle</property>
+                                        <property name="visible">True</property>
+                                        <property name="can-focus">True</property>
+                                        <property name="receives-default">False</property>
+                                        <property name="tooltip-text" translatable="yes">Directly map memory in the host address space. Much faster JIT compilation and execution.</property>
```
Comment (riperiperi) [inline Ryujinx/Ui/Windows/SettingsWindow.glade:205]: This one too.

File: Ryujinx/Ui/Windows/SettingsWindow.glade [hunk 3]
```diff
@@ -161,19 +160,86 @@
...
+                                        <property name="position">4</property>
+                                      </packing>
+                                    </child>
+                                    <child>
+                                      <object class="GtkRadioButton" id="_hideCursorAlways">
+                                        <property name="label" translatable="yes">Always</property>
+                                        <property name="visible">True</property>
+                                        <property name="can-focus">True</property>
+                                        <property name="receives-default">False</property>
+                                        <property name="tooltip-text" translatable="yes">Directly map memory, but do not mask the address within the guest address space before access. Faster, but at the cost of safety. The guest application can access memory from anywhere in Ryujinx, so only run programs you trust with this mode.</property>
```
Comment (riperiperi) [inline Ryujinx/Ui/Windows/SettingsWindow.glade:224]: Same here.

File: Ryujinx.Ava/UI/Views/Settings/SettingsUIView.axaml [hunk 4]
```diff
@@ -37,9 +37,25 @@
                     <CheckBox IsChecked="{Binding ShowConfirmExit}">
                         <TextBlock Text="{locale:Locale SettingsTabGeneralShowConfirmExitDialog}" />
                     </CheckBox>
-                    <CheckBox IsChecked="{Binding HideCursorOnIdle}">
-                        <TextBlock Text="{locale:Locale SettingsTabGeneralHideCursorOnIdle}" />
-                    </CheckBox>
+                    <StackPanel Orientation="Horizontal">
+                        <TextBlock VerticalAlignment="Center"
+                                   Text="{locale:Locale SettingsTabGeneralHideCursor}"
+                                   ToolTip.Tip="{locale:Locale MemoryManagerTooltip}"
```
Comment (riperiperi) [inline Ryujinx.Ava/UI/Views/Settings/SettingsUIView.axaml:43]: Memory manager tooltip here as well?

**Comment by riperiperi** [comment]
image
The current look of the dropdown isn't very pretty. Perhaps there needs to be a gap between this option and the checkboxes, like GTK has.

Speaking of, the Always option doesn't appear to work on GTK. When selected, the cursor is never hidden...

The always behaviour on Avalonia is a bit more aggressive than I'd like too, it hides the cursor when you're over the menu or status bars so it's hard to click them. On idle does this too, but it's not as bad as you can still see the cursor when selecting stuff. GTK's on idle mode doesn't hide the cursor unless it's over the game area, so maybe avalonia can be limited in a similar way.

**Comment by TSRBerry** [comment]
I don't know why, but I always look forward to your reviews. Thank you very much! :D

I'll try again.

**Comment by TSRBerry** [comment]
Ready for a new review!

**Review by riperiperi — APPROVED** [review]
LGTM, works like expected now.

**Comment by jojodunet** [comment]
@riperiperi thanks for the polish

**Review by AcK77 — APPROVED** [review]

**Pull request merged**